### ISSUE TITLE
[question] Is the "unique_id" for HASS meant to include NodeId?

### ISSUE THREAD

**Issue by LordMike** [OpenZWave/Zwave2Mqtt#341]
I've been playing for a few days, with the Z2M HASS integration, which works awesomely .. but once I started moving devices around in the node id's (excluding+including a few times), I got duplicate devices in HASS. 

I can see the Unique Id of the HASS discovery document uses the node Id in addition to the device type. In the below picture, my doorbell is on NodeId 2. 

![image](https://user-images.githubusercontent.com/1027111/78307154-9028af00-7545-11ea-9a53-d00a55ffe9c0.png)

I've set the "Use node names instead of NodeId" setting on the gateway, which works for the topics, but apparently not for the discovery document.

![image](https://user-images.githubusercontent.com/1027111/78307189-ac2c5080-7545-11ea-896d-824ee50884f8.png)

Is this intended?

**Comment by j3kestrel** [comment]
It is my understanding that HASS wants this to be a very unique value.  This is a policy that I find infuriating because it makes it impossible to replace a sensor without an enormous about of reconfiguration to point everything at the new sensor.  ...or what I actually do which is go to battle with HASS where I delete all the old entities and then rename the new ones to match the old ones.  For my DIY devices I don't follow HASS's intent so that I can easily swap out sensors.

I would vote for an option in zwave2mqtt to do the same.

**Comment by robertsLando** [comment]
@LordMike That is on purpose, you should use replacefailed node instead of excluede and include back your devices (to keep the same nodeid). 

Unfortunally I have moved to that unique_id after some users told me there where the same problem when doing it using node names, so after some discussion with them we decided to use that unique id that is for sure unique even if you have multiple instances of z2m running

**Comment by LordMike** [comment]
I can see the argumentation of doing it as it is now - I wrote because I encountered the same issue as @j3kestrel. 

Might I suggest then, @robertsLando, that you display the names of the nodes - even when they're not present?

As it is now. The name of a node in Z2M persists across exclude/include (by chance or by design) - which is convenient. But it requires me to remember more stuff (maybe I get sidetracked before including again). If you displayed the "name" as it was, when toggling to show "removed nodes", then I could more quickly find the node I'm meant to replace (and be reminded that I might want to replace rather than append a new node).
